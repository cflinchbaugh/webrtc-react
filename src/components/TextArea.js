import React from 'react';
import PropTypes from 'prop-types';

function TextArea(props) {
    const data = {
            id: props.id,
            value: props.value,
            onChange: (e) => {props.handleChangeValue(e.currentTarget.value)}
        }

    return <textarea {...data}/>
}

TextArea.propTypes = {
    id: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    handleChangeValue: PropTypes.func.isRequired
}

export default TextArea;