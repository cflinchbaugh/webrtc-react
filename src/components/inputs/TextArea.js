import React from 'react';
import PropTypes from 'prop-types';

function TextArea(props) {
    const data = {
            id: props.id,
            value: props.value,
            disabled: props.disabled,
            onChange: (e) => {props.handleChangeValue(e.currentTarget.value)},
        }

    return <textarea {...data}/>
}

TextArea.defaultProps = {
    disabled: false
}

TextArea.propTypes = {
    id: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    handleChangeValue: PropTypes.func.isRequired
}

export default TextArea;