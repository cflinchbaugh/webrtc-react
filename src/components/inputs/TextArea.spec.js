import React from 'react';
import { shallow } from 'enzyme';
import TextArea from './TextArea';

describe('TextArea', () => {
    it('matches snapshot', () => {
        const data = {
            id: 'id',
            value: 'value',
            handleChangeValue: jest.fn()
        }
        
        expect(shallow(<TextArea {...data}/>)).toMatchSnapshot();
    });
});
