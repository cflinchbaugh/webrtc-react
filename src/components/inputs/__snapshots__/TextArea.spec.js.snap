// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    disabled={false}
    handleChangeValue={[MockFunction]}
    id="id"
    value="value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "disabled": false,
      "id": "id",
      "onChange": [Function],
      "value": "value",
    },
    "ref": null,
    "rendered": null,
    "type": "textarea",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "disabled": false,
        "id": "id",
        "onChange": [Function],
        "value": "value",
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
